src/components/encounter/encounter-form-manager.ts

```diff
@@ -99,60 +95,51 @@ export class EncounterFormManager {
 
   static savePatientIdentifiers(patient: fhir.Patient, identifiers: PatientIdentifier[]) {
     return identifiers.map((patientIdentifier) => {
-      const identifier = getPatientLatestIdentifier(patient, patientIdentifier.identifierType);
-      if (identifier) {
-        patientIdentifier.uuid = identifier.id;
-      }
       return savePatientIdentifier(patientIdentifier, patient.id);
     });
   }
 }
 
 // Helpers
 
-function prepareObs(
-  obsForSubmission: OpenmrsObs[],
-  obsGroupsToVoid: OpenmrsObs[],
-  fields: FormField[],
-  patient: fhir.Patient,
-  encounterDate: Date,
-  encounterLocation: any,
-) {
+function prepareObs(obsForSubmission: OpenmrsObs[], fields: FormField[]) {
   fields
-    .filter((field) => field.value || field.type == 'obsGroup') // filter out fields with empty values except groups
-    .filter((field) => !field.isParentHidden && !field.isHidden && (field.type == 'obs' || field.type == 'obsGroup'))
-    .filter((field) => !field['groupId']) // filter out grouped obs
-    .filter((field) => !field.questionOptions.isTransient && field.questionOptions.rendering !== 'file')
+    .filter((field) => hasSubmitableObs(field))
     .forEach((field) => {
+      if ((field.isHidden || field.isParentHidden) && field.meta.previousValue) {
+        const valuesArray = Array.isArray(field.meta.previousValue)
+          ? field.meta.previousValue
+          : [field.meta.previousValue];
+        addObsToList(
+          obsForSubmission,
+          valuesArray.map((obs) => voidObs(obs)),
+        );
+        return;
+      }
       if (field.type == 'obsGroup') {
-        const obsGroup = {
-          person: patient?.id,
-          obsDatetime: encounterDate,
-          concept: field.questionOptions.concept,
-          location: encounterLocation,
-          order: null,
-          groupMembers: [],
-          uuid: field.uuid,
-          voided: false,
-        } as any as OpenmrsObs;
-
-        let hasValue = false;
+        if (field.meta.submission?.voidedValue) {
+          addObsToList(obsForSubmission, field.meta.submission.voidedValue);
+          return;
+        }
+        const obsGroup = constructObs(field, null) as any as OpenmrsObs;
```

Comment:
It seems bad that this type of casting is required. At the very least, always use `as unknown as ...` for these constructs. (`any` effectively says "bail out of the type system altogether"; `unknown` is "I don't know what this is").